src/main/scala/org/scalacheck/Gen.scala

```diff
@@ -784,78 +794,132 @@ object Gen extends GenArities with GenVersionSpecific {
 
   //// Character Generators ////
 
+  private def charSample(cs: Array[Char]): Gen[Char] =
+    new Gen[Char] {
+      def doApply(p: P, seed0: Seed): Gen.R[Char] = {
+        val (x, seed1) = seed0.long
+        val i = ((x & Long.MaxValue) % cs.length).toInt
```

Comment:
this is biased. Do you care? It won't produce evenly over `cs.length` since `x` is uniform over longs.

Reply:
The bias is very slight, since we have an 31-bit `Int` denominator and a 63-bit `Long` numerator. In the worst case we'll produce some values with one billionth more frequency than others. I can add a comment, but it seemed more straightforward not to add in the retry logic for this case (as opposed to `Int % Int` where the bias can be significant in the worst case).